
The following modules were not unloaded:
   (Use "module --force purge" to unload all):

  1) XALT/minimal   2) slurm   3) NeSI
2024-03-18 19:19:12.974584: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-03-18 19:19:12.975951: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-03-18 19:19:12.975999: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-03-18 19:19:12.988210: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-18 19:19:17.315333: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1886] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38298 MB memory:  -> device: 0, name: NVIDIA A100-PCIE-40GB, pci bus id: 0000:05:00.0, compute capability: 8.0
1 Physical GPUs, 1 Logical GPUs
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 conv1d (Conv1D)             (None, 16384, 8)          2056      
                                                                 
 conv1d_1 (Conv1D)           (None, 4096, 8)           8200      
                                                                 
 conv1d_transpose (Conv1DTr  (None, 16384, 8)          8200      
 anspose)                                                        
                                                                 
 conv1d_transpose_1 (Conv1D  (None, 65536, 8)          8200      
 Transpose)                                                      
                                                                 
 conv1d_transpose_2 (Conv1D  (None, 65536, 2)          18        
 Transpose)                                                      
                                                                 
 activation (Activation)     (None, 65536, 2)          0         
                                                                 
=================================================================
Total params: 26674 (104.20 KB)
Trainable params: 26674 (104.20 KB)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________
#################################
####DATA GENERATOR PARAMETERS####
#Batch size:  8
#Time in years: 0.020766743453288227
#n_channels:  2
#dt:  10
#Length of timeseries: 65536
#################################
Traceback (most recent call last):
  File "/scale_wlg_persistent/filesets/project/uoa00195/EMRI_generation/EMRI_signal_detection/train_CNN_on_EMRIs.py", line 116, in <module>
    history = model.fit(training_and_validation_generator, epochs=epochs, validation_data= training_and_validation_generator, verbose=2, callbacks=[TestOnNoise])#,use_multiprocessing=True, workers=6)
  File "/nesi/project/uoa00195/conda_envs/few_env_py3_10/lib/python3.10/site-packages/keras/src/utils/traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/scale_wlg_persistent/filesets/project/uoa00195/EMRI_generation/EMRI_signal_detection/EMRI_generator_TDI.py", line 123, in __getitem__
    X, y = self.__data_generation(indexes)#list_IDs_temp
  File "/scale_wlg_persistent/filesets/project/uoa00195/EMRI_generation/EMRI_signal_detection/EMRI_generator_TDI.py", line 150, in __data_generation
    X[batch_index,:,:]= self.noise_whiten_AET(noisy_signal_AET, self.dt, channels=self.channels_dict[self.TDI_channels])#
  File "cupy/_core/core.pyx", line 1480, in cupy._core.core._ndarray_base.__array__
TypeError: Implicit conversion to a NumPy array is not allowed. Please use `.get()` to construct a NumPy array explicitly.
